flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/read/HiveTableInputFormat.java

```diff
@@ -164,6 +186,41 @@ public void open(HiveTableInputSplit split) throws IOException {
         currentReadCount = 0L;
     }
 
+    private EnrichedRowData createEnrichedRow(HiveTableInputSplit inputSplit) {
+        // Fill the partition columns row
+        List<FileInfoExtractor.PartitionColumn> partitionColumns =
+                fileInfoExtractor.getPartitionColumns();
+        final GenericRowData fileInfoRowData = new GenericRowData(partitionColumns.size());
+        int fileInfoRowIndex = 0;
+        if (!partitionColumns.isEmpty()) {
+            final Map<String, String> partitionSpec =
+                    inputSplit.getHiveTablePartition().getPartitionSpec();
+            for (int partitionFieldIndex = 0;
+                    fileInfoRowIndex < fileInfoRowData.getArity();
+                    fileInfoRowIndex++, partitionFieldIndex++) {
+                FileInfoExtractor.PartitionColumn partition =
+                        partitionColumns.get(partitionFieldIndex);
+                if (!partitionSpec.containsKey(partition.fieldName)) {
+                    throw new RuntimeException(
+                            "Cannot find the partition value from path for partition: "
+                                    + partition.fieldName);
+                }
+
+                Object partitionValue =
+                        partitionFieldExtractor.extract(
+                                inputSplit,
+                                partition.fieldName,
+                                partition.dataType.getLogicalType());
+
+                fileInfoRowData.setField(
```

Comment:
Why not use `PartitionPathUtils#convertStringToInternalValue`? Then we don't need to call method `partition.converter.toInternal`.